vividus-engine/src/main/java/org/vividus/IdenticalStoryNamesResolver.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2019-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus;
+
+import java.io.File;
+import java.net.URI;
+import java.nio.file.InvalidPathException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.commons.io.FilenameUtils;
+import org.jbehave.core.model.Story;
+
+public class IdenticalStoryNamesResolver
+{
+    private static final char UNIX_PATH_SEPARATOR = '/';
+
+    public void resolveIdenticalNames(List<Story> stories)
+    {
+        for (Story story : stories)
+        {
+            List<Story> sameNameStories = stories.stream()
+                    .filter(s -> story.getName().equals(s.getName())).collect(Collectors.toList());
```

Comment:
```suggestion
            var storyName = story.getName();
            List<Story> sameNameStories = stories.stream()
                    .filter(s -> storyName.equals(s.getName())).collect(Collectors.toList());
```

Reply:
no `var` in production code (yet)

Reply:
yeah, sorry, just wasn't sure about the type